zerver/tests/test_link_embed.py

```diff
@@ -667,31 +667,38 @@ def test_link_preview_non_html_data(self) -> None:
         user = self.example_user("hamlet")
         self.login_user(user)
         url = "http://test.org/audio.mp3"
-        with mock_queue_publish("zerver.actions.message_send.queue_json_publish") as patched:
+        with mock_queue_publish("zerver.actions.message_send.queue_json_publish"):
             msg_id = self.send_stream_message(user, "Denmark", topic_name="foo", content=url)
-            patched.assert_called_once()
-            queue = patched.call_args[0][0]
-            self.assertEqual(queue, "embed_links")
-            event = patched.call_args[0][1]
 
-        content_type = "application/octet-stream"
-        self.create_mock_response(url, content_type=content_type)
+        msg = Message.objects.select_related("sender").get(id=msg_id)
+        event = {
+            "message_id": msg_id,
+            "urls": [url],
+            "message_realm_id": msg.sender.realm_id,
+            "message_content": url,
+        }
+
+        mocked_data = UrlOEmbedData(
+            type="audio",
+        )
+
+        self.create_mock_response(url)
 
         with self.settings(TEST_SUITE=False):
             with self.assertLogs(level="INFO") as info_logs:
-                FetchLinksEmbedData().consume(event)
-                cached_data = cache_get(preview_url_cache_key(url))[0]
+                with mock.patch(
+                    "zerver.worker.queue_processors.url_preview.get_link_embed_data",
+                    lambda *args, **kwargs: mocked_data,
+                ):
+                    FetchLinksEmbedData().consume(event)
             self.assertTrue(
                 "INFO:root:Time spent on get_link_embed_data for http://test.org/audio.mp3: "
                 in info_logs.output[0]
             )
 
-        self.assertIsNone(cached_data)
-        msg = Message.objects.select_related("sender").get(id=msg_id)
-        self.assertEqual(
-            '<p><a href="http://test.org/audio.mp3">http://test.org/audio.mp3</a></p>',
-            msg.rendered_content,
-        )
+        msg.refresh_from_db()
+        expected_content = f"""<div class="message_inline_audio"><a href="{url}"><audio controls data-audio-original-url="{url}" preload="metadata" src="{get_camo_url(url)}" title="{url}"></audio></a></div>"""
+        self.assertEqual(expected_content, msg.rendered_content)
```

Comment:
So I think we need to not remove the preexisting test, but instead modify it and add a new test. 

I believe we still support via the `FetchLinksEmbedData` code path, handling audio objects (I guess you can test one with a different file extension but an audio content-type, so that the new code doesn't apply)

Reply:
@timabbott , I have tried leaving it, but it was throwing me an error of mocks not called ([link](https://chat.zulip.org/#narrow/stream/49-development-help/topic/backend.20test.20failed.20in.20CI.20for.20inline.20playback.20of.20audio.20files/near/1671205)). I checked if the problem is my code or the test by adjusting the test to replicate it on video (.mp4) elements rather (which also has embed data), and the same test fails.
Hence, I had to modify the test, as per the test used for the YouTube Links rendering

Reply:
Updated!